src/imagery/i.eodag/i.eodag.py

```diff
@@ -580,13 +580,84 @@ def filter_result(search_result, geometry, **kwargs):
             geometry=geometry, minimum_overlap=int(minimum_overlap)
         )
 
-    if cloud_cover:
-        search_result = search_result.filter_property(
-            operator="le", cloudCover=int(cloud_cover)
-        )
```

Comment:
Why do you remove this here?

Reply:
Because if we need to filter by clouds, we can use:
`query="cloudCover=40|lt"`  (clouds < 40)
`query="cloudCover=40|gt"`  (clouds > 40)

And so it becomes more flexible, operators wise.

Reply:
For me, this adds a layer of complexity to the most common use case. The other modules and some other tools around use clouds to pass just an interest that means less than that. I wouldn't remove that simplicity which coincides with the most common usage. 

Still, I do not fully understand how search and filter modes will work in the context of i.eodag. Will users need to run the module twice or everything will happen in the background (desired)?

Reply:
> For me, this adds a layer of complexity to the most common use case. The other modules and some other tools around use clouds to pass just an interest that means less than that. I wouldn't remove that simplicity which coincides with the most common usage.

Okay, makes sense, I will put it back.
> Still, I do not fully understand how search and filter modes will work in the context of i.eodag. Will users need to run the module twice or everything will happen in the background (desired)?
No the user doesn't need to run the module twice, it happens in the background...

Here are some pseudo diagrams of the code flow:
[searching from scratch, with options] -> [filtering] -> [output results]
[read data from Text file] -> [search with ids] -> [filtering] -> [output results]
[read data from Geojson file] -> [filtering] -> [output results]

So filtering always takes place at the end before outputting the results.

Reply:
On it. I am currently working on the manual, specifically adding examples of how to use the query options and what it is capable of. I will push the manual changes when I reach a first reasonable draft.